src/runtime/hexagon_remote/remote.cpp

```diff
@@ -229,11 +504,31 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
 
 int halide_hexagon_remote_release_kernels(handle_t module_ptr, int codeLen) {
     dlclose(reinterpret_cast<void*>(module_ptr));
-
+    FARF(LOW, "HVX_TP: Master Thread: In halide_hexagon_remote_release_kernels\n");
     if (context_count-- == 0) {
         HAP_power_request(0, 0, -1);
     }
-
+    if (thread_pool_initialized) {
```

Comment:
This should be inside the if (context_count--...) above. Note that it only powers down HVX until after all of the kernels have been released, not every time kernels get released.